File: src/main/java/no/fintlabs/dispatch/converting/InstanceFlowHeadersToInstanceHeadersEntityConvertingService.java
package no.fintlabs.dispatch.converting;

import no.fintlabs.dispatch.model.InstanceHeadersEntity;
import no.fintlabs.flyt.kafka.headers.InstanceFlowHeaders;
import org.springframework.stereotype.Service;

@Service
public class InstanceFlowHeadersToInstanceHeadersEntityConvertingService {
    public InstanceHeadersEntity convert(InstanceFlowHeaders instanceFlowHeaders) {
        return InstanceHeadersEntity
                .builder()
                .sourceApplicationIntegrationId(instanceFlowHeaders.getSourceApplicationIntegrationId())
                .sourceApplicationInstanceId(instanceFlowHeaders.getSourceApplicationInstanceId())
                .archiveInstanceId(instanceFlowHeaders.getArchiveInstanceId())
                .build();
    }
}

File: src/main/kotlin/no/novari/flyt/egrunnerverv/gateway/infrastructure/dispatch/converting/InstanceFlowHeadersToInstanceHeadersEntityConvertingService.kt
package no.novari.flyt.egrunnerverv.gateway.infrastructure.dispatch.converting

import no.novari.flyt.egrunnerverv.gateway.infrastructure.dispatch.model.InstanceHeadersEntity
import no.novari.flyt.kafka.instanceflow.headers.InstanceFlowHeaders
import org.springframework.stereotype.Service

@Service
class InstanceFlowHeadersToInstanceHeadersEntityConvertingService {
    fun convert(instanceFlowHeaders: InstanceFlowHeaders): InstanceHeadersEntity =
        InstanceHeadersEntity(
            sourceApplicationInstanceId =
                instanceFlowHeaders.sourceApplicationInstanceId
                    ?: error("Missing sourceApplicationInstanceId"),
            sourceApplicationIntegrationId =
                instanceFlowHeaders.sourceApplicationIntegrationId
                    ?: error("Missing sourceApplicationIntegrationId"),
            archiveInstanceId = instanceFlowHeaders.archiveInstanceId.orEmpty(),
        )
}
